fftoml/fftoml.go

```diff
@@ -6,27 +6,78 @@ import (
 	"io"
 	"strconv"
 
-	"github.com/BurntSushi/toml"
+	"github.com/pelletier/go-toml"
 	"github.com/peterbourgon/ff"
 )
 
 // Parser is a parser for TOML file format. Flags and their values are read
 // from the key/value pairs defined in the config file.
 func Parser(r io.Reader, set func(name, value string) error) error {
-	var m map[string]interface{}
-	_, err := toml.DecodeReader(r, &m)
+	return New().Parse(r, set)
+}
+
+// ConfigFileParser is a parser for the TOML file format. Flags and their values
+// are read from the key/value pairs defined in the config file.
+// Nested tables and keys are concatenated with a delimeter to derive the
+// relevant flag name.
+type ConfigFileParser struct {
+	delimeter string
+}
+
+// New constructs and configures a ConfigFileParser using the provided options.
+func New(opts ...Option) (c ConfigFileParser) {
+	c.delimeter = "."
+	for _, opt := range opts {
+		opt(&c)
+	}
+	return c
+}
+
+// Parse parses the provided io.Reader as a TOML file and uses the provided set function
+// to set flag names derived from the tables names and their key/value pairs.
+func (c ConfigFileParser) Parse(r io.Reader, set func(name, value string) error) error {
+	tree, err := toml.LoadReader(r)
 	if err != nil {
 		return ParseError{Inner: err}
 	}
-	for key, val := range m {
-		values, err := valsToStrs(val)
-		if err != nil {
-			return ParseError{Inner: err}
+
+	return parseTree(tree, "", c.delimeter, set)
+}
+
+// Option is a function which changes the behavior of the TOML config file parser.
+type Option func(*ConfigFileParser)
+
+// TableDelimeter is an option which configures a delimeter
+// used to prefix table names onto keys when constructing
+// their associated flag name.
+// The default delimeter is "."
+func TableDelimeter(d string) Option {
```

Comment:
Cool! Just a minor nit for consistency :)

```suggestion
func WithTableDelimeter(d string) Option {
```